tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Control/AsyncType.fs

```diff
@@ -172,6 +174,31 @@ type AsyncType() =
             | _ -> reraise()
         Assert.IsTrue (t.IsCompleted, "Task is not completed")
 
+    [<Test>]
+    member this.``AwaitTask ignores Async cancellation`` () =
+        let cts = new CancellationTokenSource()
+        let tcs = new TaskCompletionSource<unit>()
+        let innerTcs = new TaskCompletionSource<unit>()
+        let a = innerTcs.Task |> Async.AwaitTask
+
+        Async.StartWithContinuations(a, tcs.SetResult, tcs.SetException, ignore >> tcs.SetCanceled, cts.Token)
```

Comment:
This looks a bit bogus to me you await the task here and use setresult as continuation. But if the task finished setresult is already called and the continuation will throw

Reply:
Ah you have two tcs so this is basically just StartAsTask or very similar

Reply:
tcs vs innerTcs, this is setting the tcs result, while the task that we AwaitTask on is from innerTcs.